tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs

```diff
@@ -73,15 +89,68 @@ public class Cpp : LanguageBase
             {
                 OperationsPerSecond = opsPerSecond,
                 StandardOutput = result.StandardOutput,
-                StandardError = result.StandardError
+                StandardError = result.StandardError,
+                PackageVersions = packageVersions
             };
         }
 
-        public override Task CleanupAsync(string project)
+        public override async Task CleanupAsync(string project)
         {
             var buildDirectory = Path.Combine(WorkingDirectory, _buildDirectory);
             Util.DeleteIfExists(buildDirectory);
-            return Task.CompletedTask;
+            await Util.RunAsync(
+                "git", $"checkout main",
+                WorkingDirectory);
+            return;
+        }
+
+        private async Task<ProcessResult> UpdatePackageVersion(string project, string packageVersion)
```

Comment:
I'm not sure this is the correct approach.  It appears you are syncing the repo to an older tag, which I don't think will work for several reasons:

1. We want to use the perf tests from main, but use older versions of the SDK packages.
2. We want the ability to mix-and-match package versions.  Say Storage in main and Core from an older package, or vice versa, or both Storage and Core from older packages.

Reply:
That wwouldn't work for cpp , the code is static built thus we cannot load dlls at runtime how we like.

Reply:
I understand (at a high level) how vcpkg works, and I think we should be able to make the C++ perf apps support arbitrary package versions, the same way a customer app can choose arbitrary versions of our packages, right?  This may require re-architecting how the perf apps are written and compiled in our repo.

Reply:
i was thinking about it i i came to the same realization. i'll revert the branch checkout part

Reply:
all we have to do is generate this https://github.com/microsoft/vcpkg/issues/21357